### PR TITLE
ECHOES-493 Add check on .d.ts generation

### PR DESCRIPTION
**Pull request by david-cho-lerat-sonarsource**
[ECHOES-493](https://sonarsource.atlassian.net/browse/ECHOES-493)

[ECHOES-493]: https://sonarsource.atlassian.net/browse/ECHOES-493?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -110,6 +112,7 @@
     "react-dom": "18.3.1",
     "react-intl": "6.7.0",
     "react-router-dom": "6.26.2",
+    "rimraf": "6.0.1",
```
Comment (gregaubert) [inline package.json:115]: Let's also update https://github.com/SonarSource/echoes-react/blob/main/package.json.md to explain why this is needed and used.

**Review by gregaubert — APPROVED** [review]
Well done! Just two possible improvements!

File: package.json [hunk 2]
```diff
@@ -35,7 +35,9 @@
     "lint-report": "eslint --ext js,ts,tsx,snap -f json -o build-reports/eslint-report/eslint-report.json .",
     "storybook": "storybook dev -p 6006",
     "tokens-check": "./config/scripts/validate-tokens.sh",
-    "ts-check": "tsc --noEmit",
+    "ts-check": "yarn ts-check-files && yarn ts-check-declarations",
+    "ts-check-files": "tsc --noEmit",
+    "ts-check-declarations": "tsc --noEmit false --emitDeclarationOnly --isolatedModules false --outFile ./build-reports/all-declarations && rimraf ./build-reports/all-declarations.*",
```
[the remark this answers is not in the document]
Comment (david-cho-lerat-sonarsource) [reply package.json:40]: yes, there are two: a `.d.ts` and a `.d.ts.map`. I'm not a fan of keeping them, as they're really just a by-product of running that check script. So I don't think a folder would be needed either. @jeremy-davis-sonarsource WDYT?

[the remark this answers is not in the document]
Comment (gregaubert) [reply package.json:40]: For me the painful part is that we have to add another node_modules to delete something that we don't have to

[the remark this answers is not in the document]
Comment (david-cho-lerat-sonarsource) [reply package.json]: OK, fair enough, I've removed `rimraf`

**Review by gregaubert — APPROVED** [review]

**Pull request merged**